src/utils/isModelValueType.ts

```diff
@@ -0,0 +1,11 @@
+import { ModelValueType } from '../glossary'
+
+export function isModelValueType(value: any): value is ModelValueType {
```

Comment:
Hmm, when I do that I get the following error message: 
`A type predicate's type must be assignable to its parameter's type.`

I think I can tweak the logic to make it work for any value, then the type is valid as `any`. What do you think about this? 

```typescript
function isPrimitiveValueType(value: any): value is PrimitiveValueType {
  return (
    typeof value === 'string' ||
    typeof value === 'number' ||
    typeof value === 'boolean' ||
    value?.constructor?.name === 'Date'
  )
}

export function isModelValueType(value: any): value is ModelValueType {
  return (
    isPrimitiveValueType(value) ||
    (Array.isArray(value) && value.every(isPrimitiveValueType))
  )
}
```

I can't think of a case where this would cause a false positive but I could be missing something.